src/inmanta/data/__init__.py

```diff
@@ -2937,6 +2937,75 @@ async def get_deleted_resources(cls, environment, current_version, current_resou
 
         return should_purge
 
+    @classmethod
+    async def get_resource_details(cls, env: uuid.UUID, resource_id: m.ResourceIdStr) -> Optional[m.ResourceDetails]:
+        status_subquery = """
+        (CASE WHEN
+            (SELECT resource.model < MAX(configurationmodel.version)
+            FROM configurationmodel
+            WHERE configurationmodel.released=TRUE
+            AND environment = $1)
+            THEN 'orphaned'
+        ELSE resource.status::text END)
+        as status
+        """
+
+        query = f"""
+        SELECT DISTINCT ON (resource_id) first.resource_id, cm.date as first_generated_time,
+        first.model as first_model, latest.resource_id as latest_resource_id, latest.resource_type,
+        latest.agent, latest.resource_id_value, latest.last_deploy as latest_deploy, latest.attributes, latest.status,
+        (SELECT JSON_OBJECT_AGG(substring(req.requires from '(.*),v='), s.status) as requires_status
```

Comment:
Why is this select required? The `requires_status` is never consumed anywhere, is it?

Reply:
You can ignore this one